govcd/vm.go

```diff
@@ -77,6 +77,22 @@ func (vm *VM) Refresh() error {
 	return err
 }
 
+// GetVirtualHardwareSection returns the virtual hardware items attached to a VM
+func (vm *VM) GetVirtualHardwareSection() (*types.VirtualHardwareSection, error) {
+
+	virtualHardwareSection := &types.VirtualHardwareSection{}
+
+	if vm.VM.HREF == "" {
+		return virtualHardwareSection, fmt.Errorf("cannot refresh, invalid reference url")
```

Comment:
Not a major thing, but we aim to return `nil` instead of empty struct because it is easier in case we check something.
Excerpt from https://github.com/vmware/go-vcloud-director/blob/master/CODING_GUIDELINES.md:
> These methods return a pointer to the entity's structure and a nil error when the search was successful, **a nil pointer and an error in every other case**.


```suggestion
		return nil, fmt.Errorf("cannot refresh, invalid reference url")
```

Reply:
@frenchtoasters Please make this change - from my perspective it's the last remaining nit. Thanks!!